tests/test_core.py

```diff
@@ -955,6 +956,21 @@ def test_is_allowed_path(self):
             with pytest.raises(AssertionError):
                 self.config.is_allowed_path(None)
 
+    def test_is_allowed_external_url(self):
+        """Test is_allowed_external_url method."""
+        self.config.allowlist_external_urls = [
+            "http://x.com",
+            "https://y.com",
+        ]
+
+        valid = ["http://x.com/1.jpg", "https://y.com/2.png"]
+        for url in valid:
+            assert self.config.is_allowed_external_url(url)
+
+        invalid = ["http://z.com/stream", "https://a.co"]
```

Comment:
Can you add a test with paths and verify that we do not allow `..` ? 

ie allow `http://x.com/bla/` and make sure `http://x.com/bla/../something` is not allowed.

Reply:
You can use the URL lib `yarl`, as it is part of `aiohttp` and so installed.

Reply:
```python
>>> import yarl
>>> yarl.URL("http://x.com/bla/../something")
URL('http://x.com/something')
```